src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/model/SwitchExtended.groovy

```diff
@@ -0,0 +1,677 @@
+        def blockData = lockKeeper.knockoutSwitch(sw, regions)
+        cleanupManager.addAction(REVIVE_SWITCH, { reviveSwitch(blockData, true) }, CleanupAfter.TEST)
+        return blockData
+    }
+
+    List<FloodlightResourceAddress> knockoutSwitchFromStatsController(){
+        def blockData = lockKeeper.knockoutSwitch(sw, FloodlightConnectMode.RO)
+        cleanupManager.addAction(REVIVE_SWITCH, { reviveSwitch(blockData, true) }, CleanupAfter.TEST)
+        return blockData
+    }
+
+    /**
+     * Connect a switch to controller either adding controller settings inside an OVS switch
+     * or setting proper iptables to allow access to floodlight for a hardware switch.
+     *
+     * @param flResourceAddress to register sw in the specific FL regions
+     * @param waitForRelatedLinks make sure that all switch related ISLs are DISCOVERED
+     */
+    void reviveSwitch(List<FloodlightResourceAddress> flResourceAddress, boolean waitForRelatedLinks) {
+        lockKeeper.reviveSwitch(sw, flResourceAddress)
+        Wrappers.wait(WAIT_OFFSET) {
+            assert northbound.getSwitch(sw.dpId).state == SwitchChangeType.ACTIVATED
+        }
+        if (waitForRelatedLinks) {
+            Wrappers.wait(KildaProperties.DISCOVERY_TIMEOUT + WAIT_OFFSET * 2) {
+                def allIsls = northbound.getAllLinks()
+                isls.collectMany { [it, it.reversed] }.each { isl ->
+                    assert allIsls.find {
+                        (it.source.switchId == isl.srcSwitch.dpId && it.source.portNo == isl.srcPort
+                                && it.destination.switchId == isl.dstSwitch.dpId && it.destination.portNo == isl.dstPort)
+                    }.state == IslChangeType.DISCOVERED
+                }
+            }
+        }
+    }
+
+    void reviveSwitch(List<FloodlightResourceAddress> flResourceAddress) {
+        reviveSwitch(flResourceAddress, false)
+    }
+
+
+    /**
+     * This method calculates expected burst for different types of switches. The common burst equals to
+     * `rate * BURST_COEFFICIENT`. There are couple exceptions though:<br>
+     * <b>Noviflow</b>: Does not use our common burst coefficient and overrides it with its own coefficient (see static
+     * variables at the top of the class).<br>
+     * <b>Centec</b>: Follows our burst coefficient policy, except for restrictions for the minimum and maximum burst.
+     * In cases when calculated burst is higher or lower of the Centec max/min - the max/min burst value will be used
+     * instead.
+     *
+     * @param rate meter rate which is used to calculate burst
+     * Needed to get the switch manufacturer and apply special calculations if required
+     * @return the expected burst value for given switch and rate
+     */
+    def retrieveExpectedBurst(long rate) {
+        def burstCoefficient = KildaProperties.BURST_COEFFICIENT
+        if (isNoviflow() || isWb5164()) {
```

Comment:
I guess, the beautiful solution here would be inherit `NoviflowSwitchExtended`, `WbSwitchExtended`, etc. with overwrittend `getExpectedBurst()` method, but it seems to be too complicated for just one method.

Reply:
agreed, let's keep the current implementation